azuredevops/resource_azure_git_repository.go

```diff
@@ -0,0 +1,145 @@
+package azuredevops
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/microsoft/azure-devops-go-api/azuredevops/core"
+	"github.com/microsoft/azure-devops-go-api/azuredevops/git"
+	"github.com/microsoft/terraform-provider-azuredevops/azuredevops/utils/converter"
+)
+
+func resourceAzureGitRepository() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAzureGitRepositoryCreate,
+		Read:   resourceAzureGitRepositoryRead,
+		Update: resourceAzureGitRepositoryUpdate,
+		Delete: resourceAzureGitRepositoryDelete,
+
+		//https://godoc.org/github.com/hashicorp/terraform/helper/schema#Schema
+		Schema: map[string]*schema.Schema{
+			"project": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				ForceNew: false,
+				Required: true,
+			},
+			"default_branch": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "",
+			},
+			"is_fork": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"remote_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"size": {
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+			"ssh_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"web_url": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"valid_remote_urls": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+		},
+	}
+}
+
+func resourceAzureGitRepositoryCreate(d *schema.ResourceData, m interface{}) error {
+	name := d.Get("name").(string)
+
+	d.SetId(name)
+
+	return resourceAzureGitRepositoryRead(d, m)
+}
+
+func resourceAzureGitRepositoryRead(d *schema.ResourceData, m interface{}) error {
+	clients := m.(*aggregatedClient)
+
+	id := d.Id()
+	name := d.Get("name").(string)
+	projectNameOrId := d.Get("project").(string)
+	repo, err := azureGitRepositoryRead(clients, id, name, projectNameOrId)
+	if err != nil {
+		return fmt.Errorf("Error looking up repository with ID %s and Name %s", id, name)
```

Comment:
We loose the context of the original error here. A suggestion would be to do the following:

```suggestion
		return fmt.Errorf("Error looking up repository with ID %s and Name %s. Error: %v", id, name, err)
```